core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2131,15 +2138,20 @@ class Log(@volatile var dir: File,
    *
    * @throws IOException if the file can't be renamed and still exists
    */
-  private def asyncDeleteSegment(segment: LogSegment) {
-    segment.changeFileSuffixes("", Log.DeletedFileSuffix)
-    def deleteSeg() {
-      info(s"Deleting segment ${segment.baseOffset}")
+  private def deleteSegmentFiles(segments: List[LogSegment], asyncDelete: Boolean) {
+    info(s"Deleting segments $segments")
```

Comment:
It's probably better to log the segment deletion just before it's actually deleted. If we want to add the extra logging at the time of scheduling, it would be fine.

Reply:
Hmm, I thought physical deletion of segments is not interesting and perhaps we could log logical deletions only. I will revert the logging to how it was before this change.